### PR TITLE
fix public key out put in generateKeys

### LINKED ISSUE #15: sam3GenerateKeys returns wrong public key
The returned public key is the end of the private key:

```
c++ test.cpp -lsam3 -L`pwd` && ./a.out
pub  qex~CbtDpHmoGoPK20SMg==
priv 8OiN7vVSRL6TMxjFdBHOq~IFwOfMso3QnGzeQ8ASJfI6EqMR4KwUC5AA4Klic4bciqKb7UmoesHTLT7kGw5UPBTlc1xGcagB-M39l7M3LXVK8~prD6AMXDw-b4hg006nzvhfvc6TdqZ2JyrTqAvIMDvdxg0GKRRyzdU1Nweol1vwRjSM5UkheKJAxcY9~lIWaFpKYNhpdR8Kdsm29~9OsMBNt31ejLJ5ccdtPvR5nUrDvpKDHXxunrRhEBOq5O02TvmeGQv5qhwUC5hFUP8w2ZQ~7LGCCY~QTqRl2ttW~nFgJakMxS0w7UPKkvQxmV-k74DW82RZfb-l6Dpr-RtTdL6ysaaWo1JjB6CZfUypanMw7nleLX~nCPGVWalAv2xyEUbi14Pcs3KHo~2tXQ35ypSJvem~uvUE7OGgAxuadAVz9~GafI3hA3lda76IKZOVFiz8s46~xSl9ES2GalvBuLRxJPCEzoodMxPAyacmgTOtG~f15WrCE5LJE15Vg73gBQAEAAcAAKuzeJdCuBl2am432oZdCYNsvVW8aOz7pLfFVZ-W27V2PykNOdgDLqlrGBmwl4aEKev5oXfQMiZGJCUmxhAnxxOAa1lNHrKhDqAe2cEePMiq8X-QCf4XbhIc9EQ8y6mvNmOUDN4KueR5pgpft2zt1zCgIBRmIzaqUN86Oc8mcMIWRT8x9WM4T1HhoMAjjNz-R6duCaJTmZD5bZhBkk4V95mD3KZF8EqXKL2IIUEVQgdG8oT9AjkTvc0znmjXCuxqt1qzjoLZnQuAOW7Z2ACUxhv~LAaJzS3cb-ksirVR8TKlbRxCpKrckso550tv4Xw6-ETGaapJJZZ9BZAIvpQ8U6oLMrsGCGHbGPp0~VfIVS3uQqex~CbtDpHmoGoPK20SMg==
```

Code
```c++
#include <string>
#include <iostream>
#include <stdio.h>
#include "src/libsam3/libsam3.h"

int main() {
	// The session is only usef for transporting the data
	Sam3Session ss;

	if (0 > sam3GenerateKeys(&ss, SAM3_HOST_DEFAULT, SAM3_PORT_DEFAULT, Sam3SigType::EdDSA_SHA512_Ed25519)) {
		printf("got error");
		return -1;
	}
	auto pub = std::string(ss.pubkey);
	auto priv = std::string(ss.privkey);

	std::cout << "pub  " << pub << std::endl << "priv " << priv << std::endl;
	return 0;
} 
```

### PR DESCRIPTION
**Pull request by eyedeekay**
Fixes #15

### CODE REVIEW

**Comment by eyedeekay** [comment]
@sehraf This should fix the erroneous public key output. If you have time, please check if it works for you.

**Comment by sehraf** [comment]
Works for me with my reproducer from the issue.

**Comment by eyedeekay** [comment]
Thanks I added your reproducer to the examples which I run as tests. I'll merge and cut another tag in a few minutes.

File: src/libsam3/libsam3.c [hunk 1]
```diff
@@ -716,20 +716,23 @@ int sam3GenerateKeys(Sam3Session *ses, const char *hostname, int port,
...
-                   *priv = sam3FindField(rep, "PRIV");
-        //
-        if (pub != NULL && sam3CheckValidKeyLength(pub) && priv != NULL &&
-            strlen(priv) >= SAM3_PRIVKEY_MIN_SIZE) {
-          strcpy(ses->pubkey, pub);
-          strcpy(ses->privkey, priv);
-          res = 0;
-        }
-      }
+    if (sam3tcpPrintf(fd, "DEST GENERATE\n") < 0) {
```
Comment (sehraf) [inline src/libsam3/libsam3.c:719]: Aren't you missing the sign type here?

(Sorry for pointing this out, so late)

Comment (eyedeekay) [reply src/libsam3/libsam3.c:719]: Yup, looks like in this case we would be limited to the defaults. I'll work something up to fix it.

**Pull request merged**